### PR TITLE
Order show us30

### LINKED ISSUE #30: User Story 26, Registered users can check out
As a registered user
When I add items to my cart
And I visit my cart
I see a button or link indicating that I can check out
And I click the button or link to check out and fill out order info and click create order
An order is created in the system, which has a status of "pending"
That order is associated with my user
I am taken to my orders page ("/profile/orders")
I see a flash message telling me my order was created
I see my new order listed on my profile orders page
My cart is now empty

### PR DESCRIPTION
**Pull request by leahriffell**
closes #30 

**Contributors**
- Driver(s): Leah
- Navigator(s): Priya, Luke, Grant
  
**What was changed:**
- Added ability for user to cancel an order if pending
- When cancelled, order status changes, all item orders change to unfulfilled
- Item orders that were already fulfilled should increase inventory count (couldn't get this to pass)

**Any tech notes (if applicable)**
- Couldn't get model test to pass for increasing inventory when a fulfilled item_order becomes unfulfilled

### CODE REVIEW

File: app/models/order.rb [hunk 1]
```diff
@@ -5,22 +5,32 @@ class Order <ApplicationRecord
 
   validates_presence_of :name, :address, :city, :state, :zip, :status
 
-  enum status: %w(pending packaged shipped cancelled)
+  enum status: %w(Pending Packaged Shipped Cancelled)
```
Comment (LHJE) [inline app/models/order.rb:8]: Great addition

File: app/models/order.rb [hunk 2]
```diff
@@ -5,22 +5,32 @@ class Order <ApplicationRecord
...
+        item = Item.find(item_order.item_id)
+        item.update(inventory: (item.inventory + item_order.quantity))
+      end
+    end
+  end
+
+  def cancel 
+    change_to_cancel_status
+    unfulfill_item_orders
+  end
```
Comment (LHJE) [inline app/models/order.rb:35]: So glad to have all these done!

File: spec/models/order_spec.rb [hunk 3]
```diff
@@ -24,25 +24,41 @@
...
+      # expect(@tire.inventory).to eq(17)
+      @order_2.item_orders.all?{ |item_order| expect(item_order.fulfilled?).to eq(false) }
+    end
+
+    it "cancel" do 
+      @order_2.cancel 
+      expect(@order_2.status).to eq("Cancelled")
+      # expect(@tire.inventory).to eq(17)
+      @order_2.item_orders.all?{ |item_order| expect(item_order.fulfilled?).to eq(false) }
+    end
```
Comment (LHJE) [inline spec/models/order_spec.rb:62]: We'll figure out how to implement this I'm sure of it.

**Review by LHJE — APPROVED** [review]
Great work!

File: spec/models/order_spec.rb (hunk 3 above)
Comment (leahriffell) [reply spec/models/order_spec.rb:62]: Thank you for your vote of confidence

**Pull request merged**